### PR TITLE
fix(CI): add spark profile to all spark tests

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: tests/test_gatk4spark.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@
     - path: results/preprocessing/mapped/
       should_exist: false
 - name: Run default pipeline with gatk4spark and skipping all QC steps
-  command: nextflow run main.nf -profile test_cache,use_gatk_spark --skip_tools fastqc,markduplicates_report,mosdepth,multiqc,samtools --outdir results
+  command: nextflow run main.nf -profile test_cache,use_gatk_spark,spark --skip_tools fastqc,markduplicates_report,mosdepth,multiqc,samtools --outdir results
```
Comment (FriederikeHanssen) [inline tests/test_gatk4spark.yml:52]: I missed a bunch of discussion around this. What is tl;dr that we have a double spark profile?

Comment (maxulysse) [reply tests/test_gatk4spark.yml:52]: So 1 for the test, and one for the docker.userEmulation that restore proper usage

Comment (FriederikeHanssen) [reply tests/test_gatk4spark.yml:52]: ah makes sense, so users should set `spark` when runnin it now?

Comment (FriederikeHanssen) [reply tests/test_gatk4spark.yml:52]: makes sense :)

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**